### PR TITLE
BUG: Fix incorrect deletion from addrsSorted

### PR DESCRIPTION
**Pull request by rahulghangas**
When deleting an address, the wrong index was being deleted, and if the address to be deleted was present at the end of the array, was not being deleted at all. This was leading to disparity between the `addrsBySignatory` hashmap and `addrsSorted` array. This PR fixes that and adds a correctness test to `dht_test.go`

- [x] Correctness tests (might need more)

This bug affects gossiping as well since `dht.Addrs(n)` uses  `addrsSorted`

### CODE REVIEW

**Review by jazg — DISMISSED** [review]
Good catch! It might be worth adding some explicit edge cases to the test (just because quick check might miss the first/last elements, for example), but overall looks good!

**Comment by rahulghangas** [comment]
I've added the edge cases, I think it's ready to be merged. Can someone do a sanity check on this?

File: dht/dht_test.go [hunk 1]
```diff
@@ -134,6 +134,128 @@ var _ = Describe("DHT", func() {
...
+					ok := table.InsertAddr(randAddr)
+					Expect(ok).To(BeTrue())
+				}
+
+				// Get sorted addresses and check if they are equal
+				// to the number of inserted addresses
+				addrs := table.Addrs(numAddrs)
+				Expect(len(addrs)).To(Equal(numAddrs))
+
+				// Remove last addresses, query sorted addresses again
```
Comment (jazg) [inline dht/dht_test.go:158]:
```suggestion
				// Remove first addresses, query sorted addresses again
```

**Review by jazg — DISMISSED** [review]

**Review by jazg — APPROVED** [review]

**Comment by jazg** [comment]
@rahulghangas Approved!

**Comment by rahulghangas** [comment]
Closing this, changes have been integrated into the experimental branch
